ambari-common/src/main/python/resource_management/libraries/functions/get_user_call_output.py

```diff
@@ -40,10 +41,11 @@ def get_user_call_output(command, user, quiet=False, is_checked_call=True, **cal
   try:
     out_files.append(tempfile.NamedTemporaryFile())
     out_files.append(tempfile.NamedTemporaryFile())
-    
+    uid = pwd.getpwnam(user).pw_uid
     # other user should be able to write to it
```

Comment:
At some point someone thought that the generated files should be world writable...

```
# other user should be able to write to it
```

However, looking at the function it seems like only the owner of the process executing the command needs to write to them.  The trick though is setting the correct owner when the Ambari agent does not run as root. 

@aonishuk, and comment on this?